cpp/test/cluster/linkage.cu

```diff
@@ -14,6 +14,15 @@
  * limitations under the License.
  */
 
+// XXX: We allow the instantiation of fused_l2_nn here:
```

Comment:
Am I correct in understanding that we need to implicitly instantiate `fused_l2_nn` here because `connect_components` doesn't have an extern and explicit instantiation?

This makes sense. We implicitly instantiate `connect_components` here which I imagine has a cascading effect on every single function that is called within its function body. It's using the header-only version so `RAFT_EXPLICIT` error must be thrown for `fused_l2_nn`, and it's good to see that error working as expected.

Reply:
Yes, exactly :+1: